File: Easy/342. Power of Four/342. Power of Four.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function (n) {
    if (n <= 0) {
        return false;
    }

    // Check if n is a power of 2
    if ((n & (n - 1)) === 0) {
        // Check if the logarithm of n to the base 4 is an integer
        const logResult = Math.log(n) / Math.log(4);
        return logResult % 1 === 0;
    } else {
        return false;
    }
};

File: Easy/342. Power of Four/342. Power of Four.rs
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n <= 0 {
            return false;
        }

        // Check if n is a power of 2
        if n & (n - 1) == 0 {
            // Check if the logarithm of n to the base 4 is an integer
            let log_result = (n as f64).log(4.0);
            log_result.fract() == 0.0
        } else {
            false
        }
    }
}
